components/dashboard/src/prebuilds/list/PrebuildListItem.tsx

```diff
@@ -0,0 +1,101 @@
+import { LoadingState } from "@podkit/loading/LoadingState";
+
+type Props = {
+    prebuild: Prebuild;
+};
+export const RepositoryListItem: FC<Props> = ({ prebuild }) => {
+    const created = dayjs(prebuild.status?.startTime?.toDate()).fromNow();
+    const {
+        data: configuration,
+        isError: isConfigurationError,
+        isLoading: isConfigurationLoading,
+    } = useConfiguration(prebuild.configurationId);
+    const { className: iconColorClass, label } = prebuildDisplayProps(prebuild);
+    const PrebuildStatusIcon = prebuildStatusIconName(prebuild);
+
+    return (
+        <TableRow>
+            <TableCell>
+                <div className="flex flex-col gap-1">
+                    <Text className="text-sm text-pk-content-primary text-semibold">
+                        <ConfigurationField
+                            configuration={configuration}
+                            isError={isConfigurationError}
+                            isLoading={isConfigurationLoading}
+                        />
+                    </Text>
+                    <TextMuted className="text-xs break-all">{prebuild.ref}</TextMuted>
+                </div>
+            </TableCell>
+
+            <TableCell hideOnSmallScreen>
+                {prebuild.commit?.author && (
+                    <div className="flex flex-col gap-1">
+                        <Text className="text-sm text-pk-content-secondary">
+                            {shortCommitMessage(prebuild.commit.message)}
+                        </Text>
+                        <div className="flex gap-1 items-center">
+                            <img src={prebuild.commit.author.avatarUrl} className="w-5 h-5 rounded-full" alt="" />
+                            <Text className="text-xs break-all text-pk-content-secondary">
+                                {prebuild.commit.author.name}
+                            </Text>
+                        </div>
+                    </div>
+                )}
+            </TableCell>
+
+            <TableCell hideOnSmallScreen>
+                <Text className="text-sm break-all text-pk-content-secondary">{created}</Text>
+            </TableCell>
+
+            <TableCell>
+                <div className="flex flex-row gap-1.5 items-center capitalize">
+                    <PrebuildStatusIcon className={cn("w-5 h-5", iconColorClass)} />
+                    <span className="text-sm text-pk-content-secondary">{label}</span>
+                </div>
+            </TableCell>
+
+            <TableCell>
+                <LinkButton href={`/prebuilds/${prebuild.id}`} disabled variant="secondary">
+                    View
+                </LinkButton>
+            </TableCell>
+        </TableRow>
+    );
+};
+
+type ConfigurationProps = {
+    configuration?: Configuration;
+    isLoading: boolean;
+    isError: boolean;
+};
+const ConfigurationField = ({ configuration, isLoading, isError }: ConfigurationProps) => {
```

Comment:
The biggest con of this way of conditionally rendering the field is that it takes ~ half a second after the render before this changes from loading to unknown, even with cache. We could think about including the name in the prebuild response, but that would not solve the issue of deleted configs, just make the latency easier to bear.